third_party/leiningen/lein_repo.bzl

```diff
@@ -71,4 +82,26 @@ lein_repository = repository_rule(
         ),
         "sha256": attr.string(mandatory = True),
     },
+    environ = ["PATH", "LEIN_JAVA_CMD"],
 )
+
+def _check_jdk(repository_ctx):
+    java = repository_ctx.os.environ.get("LEIN_JAVA_CMD") or repository_ctx.which("java")
+    result = repository_ctx.execute([java, "-version"])
+    if result.return_code != 0:
+        fail("Error determining JDK version: {error} (with LEIN_JAVA_CMD={java})".format(
```

Comment:
Does it still make sense to say LEIN_JAVA_CMD= when we get java from which("java")?

Reply:
IMO, yes as we are using this as the default value of that variable if it's empty or missing.  Plus it's an easy way to document that the user can specify it explicitly if they should choose to do so.